Instance details page and event data docs (#848)
replicatedhq/replicated-docs: docs/vendor/monitoring-event-data.md

    <td>string</td>
    <td>KOTS Version</td>
  </tr> 
  <tr>
    <td><code>versionLabel</code></td>
    <td>The version that the instance is running. The <code>versionLabel</code> is the version assigned to the release when the release was promoted.</td>
    <td>string</td>
    <td>App Version</td>
  </tr> 
  <tr>
    <td><code>upgradeSuccess</code></td>
    <td>Tracks when an upgrade is successful according to the defined criteria.</td>
    <td>boolean</td>
    <td><p>Upgrade Successful</p><p>Upgrade Failed</p>
</td>
  </tr> 
</table>

## Cluster Status Events

<table>
  <tr>
    <th>Field Name</th>
    <th>Description</th>
  </tr> 
  <tr>
    <td><code>isKurl</code></td>
    <td>
      <p>Indicates if the cluster was provisioned by the Replicated Kubernetes installer.</p>
      <p>Possible values:</p>
      <ul>
        <li><code>kURL</code>: The cluster is a Kubernetes installer cluster.</li>
        <li><code>Existing</code>: The cluster is <em>not</em> a Kubernetes installer cluster.</li>
      </ul>
    </td>
  </tr> 
  <tr>
    <td><code>k8sVersion</code></td>

>     <td>The version of Kubernetes running in the cluster. The Kubernetes version often contains the distribution. For example, <code>1.24.5-eks-sha</code>.</td>

Reviewer: ```suggestion
    <td>The version of Kubernetes running in the cluster. The Kubernetes version often contains details that may help determine the Kubernetes distribution. For example, <code>1.24.5-eks-ae7312d</code>.</td>
```

  </tr>
  <tr>
    <td><code>k8sDistribution</code></td>
    <td>
      <p>The Kubernetes distribution of the cluster, when detected.</p>
      <p>Possible values:</p>
      <ul>
        <li>EKS</li>
        <li>GKE</li>
        <li>K3S</li>
        <li>RKE2</li>
      </ul>
    </td>
  </tr>
  <tr>
    <td><code>kurlNodeCountTotal</code></td>

>     <td>Total number of nodes in the Kubernetes installer cluster. Applies to clusters provisioned by the Kubernetes installer only.</td>

Reviewer: ```suggestion
    <td>Total number of nodes in the Kubernetes cluster. Applies to clusters provisioned by the Kubernetes installer only.</td>
```

  </tr>
  <tr>
    <td><code>kurlNodeCountReady</code></td>

>     <td>Number of nodes in the Kubernetes installer cluster that are in a healthy state and ready to run Pods. Applies to clusters provisioned by the Kubernetes installer only.</td>

Reviewer: Kubernetes Installer can be a synonym for kURL in some cases, but in this one it's especially weird.

```suggestion
    <td>Number of nodes in the Kubernetes cluster that are in a healthy state and ready to run Pods. Applies to clusters provisioned by the Kubernetes installer only.</td>
```

Author: Is it "Kubernetes installer" in general that's sounding awkward you think?

For context, this is the terminology we're using in the docs to talk about clusters provisioned by the Kubernetes installer (rather than "embedded cluster", which isn't the most clear). So, according to our style guide, the first mention in a topic should be spelled out as "a cluster provisioned by the Replicated Kubernetes installer". And then subsequent mentions in the topic are just "Kubernetes installer cluster" as shorthand.

The other goal was to make sure we are using the commercial name (which is "Kubernetes installer") rather than the open source name (kURL). Same thing that we do using app manager/admin console rather than KOTS. The idea here is that we want the Replicated feature commercial term to stay the same, even if the underlying open source project changes. 

The fact that the Replicated features are always lowercased I think probably contributes to this sort of thing seeming awkward, though not sure if that's the sticking point for you as well

  </tr>
  <tr>
    <td><code>kurlNodeCountReady</code></td>
    <td></td>
  </tr>
  <tr>
    <td><code>kurlInstallerSpecID</code></td>
    <td></td>
  </tr>
</table>

## Infrastructure Status Events

<table>
  <tr>
    <th>Field Name</th>
    <th>Description</th>
  </tr> 
  <tr>
    <td><code>cloudProvider</code></td>
    <td>
      <p>The cloud provider where the instance is running. Determined from the IP address that makes the request.</p>

>       <p>Possible values:</p>

Reviewer: I think there are more. Maybe 

```suggestion
      <p>Possible values include:</p>
```

      <ul>
        <li>AWS</li>
        <li>GCP</li>
        <li>DigitalOcean</li>
      </ul>
    </td>
  </tr>
  <tr>
    <td><code>cloudProviderRegion</code></td>
    <td>
      <p>The cloud provider region where the instance is running. For example, <code>us-central1-b</code></p>
    </td>
  </tr>  
    <tr>
    <td><code>kurlOSFlavor</code>*</td>
    <td>One or more operating systems detected across cluster nodes</td>
  </tr>
  <tr>
    <td><code>kurlOSVersion</code>*</td>
    <td>One or more operating systems detected across cluster nodes</td>
  </tr>
</table>

## Preflight Check Events

> 

Reviewer: ```suggestion
**Note**: Preflight check events will only be sent by instances running KOTS version 1.93.x or later.
```

<table>
  <tr>
    <th>Field Name</th>
    <th>Description</th>
  </tr> 
  <tr>
    <td><code>preflightCheckStatus</code></td>
    <td>
    </td>
  </tr>
</table>

## Upstream Update Events

<table>
  <tr>
    <th>Field Name</th>
    <th>Description</th>
  </tr> 
  <tr>

>     <td><code>versionAge</code></td>

Reviewer: I don't think these actually come as events, just insights

Author: Making sure I understand "come as events" and "insights".

Do you mean that `versionAgeSinceLatest` and `versionAge` are not part of the data sent by the app manager, but rather they are metrics that are computed by the vendor portal?

Feels like this is an important distinction so this topic can be really clear about defining the event data fields that are listed in these tables (ie, we could differentiate between the stuff sent by the app manager and the metrics that are computed by the vendor portal)

Author: Clarified that these are computed by the vendor portal when the app manager sends data.

    <td>
    </td>
  </tr>
  <tr>

>     <td><code>versionAgeSinceLatest</code></td>

Reviewer: I don't think these actually come as events, just insights

Author: Clarified that these are computed by the vendor portal when the app manager sends data.

Reviewer: I guess my point is, they will never show up in the "event stream", so they're unique in this table. They're only show in tiles, not as raw events

In addition to creating events for changes in data fields sent by the app manager, the vendor portal also generates events for changes in the value of a computed metric. For example, the vendor portal computes a `numberVersionsBehind` metric that tracks the number of versions behind the latest available version for the instance. When the instance checks for updates and the vendor portal identifies a new version that is available to the instance, then the vendor portal generates an event to indicate the change in the value of the `numberVersionsBehind` metric. The vendor portal updates the values of computed metrics each time the app manager sends instance data.

Each event that the vendor portal generates for application instances has the following fields:

* `fieldName`: The instance data field that generated the event. For example, `appStatus`. 
* `previousValue`: The value of the data field before the vendor portal generated the event.
* `newValue`: The value of the data field after the vendor portal generated the event.

The vendor portal uses events to display insights for each active application instance in a **Instance details** dashboard. 

> For more information about using the vendor portal **Instance details** page to monitor active instances of your application, see [Monitoring Application Instances](monitoring-instance-details).

Reviewer 2: Need to change "see [Monitoring Application Instances](monitoring-instance-details)" to "see [Viewing Instance Details](monitoring-instance-details)"

## Instance Events

This section describes each type of event that the vendor portal generates for active application instances. Events in the vendor portal are grouped into the following categories:

* [Application Installation and Upgrade Events](#install-upgrade)
* [Cluster Status Events](#cluster)
* [Infrastructure Status Events](#infrastructure)
* [KOTS Status Events](#kots)
* [Upstream Update Events](#upstream)

The tables in this section include the following details about each event type:

* **Field Name**: The `fieldName` associated with the event.
* **Description**: A description of the data field.
* **Type**: The data type of the field. Possible values are string, number, and boolean.

> * **Label**: The label for the event that displays in the **Instance Activity** stream in the vendor portal **Instance Details** page. For more information, see [Instance Activity](monitoring-instance-details#instance-activity) in _Monitoring Application Instances_.

Reviewer 2: Need to change "see [Monitoring Application Instances](https://github.com/replicatedhq/replicated-docs/pull/848/monitoring-instance-details)" to "see [Viewing Instance Details](https://github.com/replicatedhq/replicated-docs/pull/848/monitoring-instance-details)"

    <p><strong>Note:</strong> Applies only to clusters provisioned by the Kubernetes installer.</p>
    </td>
    <td>string</td>
    <td>New kURL Installer</td>
  </tr>  
</table>

### Infrastructure Status Events {#infrastructure}

<table>
  <tr>
    <th>Field Name</th>
    <th>Description</th>
    <th>Type</th>
    <th>Label</th>
  </tr> 
  <tr>
    <td><code>cloudProvider</code></td>
    <td>
      <p>The cloud provider where the instance is running. <code>cloudProvider</code> is determined by the IP address that makes the request.</p>
      <p>Possible values:</p>
      <ul>
        <li>AWS</li>
        <li>GCP</li>
        <li>DigitalOcean</li>
      </ul>
    </td>
    <td>string</td>
    <td>Cloud Provider</td>
  </tr>
  <tr>
    <td><code>cloudProviderRegion</code></td>
    <td>

>       <p>The cloud provider region where the instance is running. For example, <code>us-central1-b</code></p>

Reviewer 2: You could change "For example," to "**Example:**

Author: I left this as is just to not mess with too much formatting in the HTML table
